### PR TITLE
Allen/timezone in log data

### PR DESCRIPTION
**Pull request by algtan**
Added a shifted timezone date when viewing an individual log. This required the following:
1. Adding the timezone information to the state of `app`
2. Passing the timezone information to `LogView` as a prop
3. Installing `luxon` to get the shifted timezone date in the proper format

NOTE: Unit testing not done at this time because it requires resolving issues with the package `react-timezone-select`, and with imports of non-JS files (e.g. PNG, CSS).

### CODE REVIEW

File: src/client/components/Sidebar.tsx [hunk 1]
```diff
@@ -126,7 +132,7 @@ class Sidebar extends PureComponent<SidebarProps, SidebarState> {
             />
             <TimezonePicker
               currentTimezone={this.state.timezone.value}
-              onChange={timezone => this.handleChange({ timezone })}
+              onChange={timezone => this.handleTimezoneChange(timezone)}
```
Comment (thehobbit85) [inline src/client/components/Sidebar.tsx]: I think that this line can just be:
```onChange={this.handleTimezoneChange}```

I may be wrong so test it.

File: src/client/components/LogView.tsx [hunk 2]
```diff
@@ -63,12 +68,21 @@ class LogView extends Component<LogViewProps, LogViewState> {
     })
   }
 
+  findShiftedTimezoneDate = (logTimestamp, timezoneName): string => {
+    const logDateUtc = new Date(logTimestamp * 1000)
+    return DateTime.fromISO(logDateUtc.toJSON(), { zone: timezoneName }).toISO()
+  }
+
```
Comment (thehobbit85) [inline src/client/components/LogView.tsx]: No need to use a function here, it's only used once, it's super simple, and you are actually just making the code longer and more convoluted this way.

File: src/client/components/LogView.tsx [hunk 3]
```diff
@@ -82,7 +96,7 @@ class LogView extends Component<LogViewProps, LogViewState> {
           onClick={() => this.setState({ collapsed: !this.state.collapsed })}
         />
         <ReactJson
-          src={this.state.log}
+          src={{ shiftedTimezoneDate: shiftedTimezoneDate, ...this.state.log }}
```
Comment (thehobbit85) [inline src/client/components/LogView.tsx]: When both the `key` and the `value` have the same value, you don't need to repeat it:
This is how it should look like:
```src={{ shiftedTimezoneDate, ...this.state.log }}```

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments

File: src/client/components/Sidebar.tsx (hunk 1 above)
Comment (algtan) [reply src/client/components/Sidebar.tsx]: Updated per comment. I also changed `handleTimezonChange` to an arrow function to preserve the context of `this`. Making the recommended change also improves optimization, as having the arrow function in the render means that a new function is created every time `render` is called.

Ref: https://daveceddia.com/avoid-bind-when-passing-props/

File: src/client/components/LogView.tsx (hunk 3 above)
Comment (algtan) [reply src/client/components/LogView.tsx]: Updated per comment.

File: src/client/components/LogView.tsx (hunk 2 above)
Comment (algtan) [reply src/client/components/LogView.tsx]: Updated per comment.

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**